synapse/handlers/message.py

```diff
@@ -1307,13 +1364,21 @@ async def persist_and_notify_client_event(
             if prev_state_ids:
                 raise AuthError(403, "Changing the room create event is forbidden")
 
+        # Mark any `m.historical` messages as backfilled so they don't appear
+        # in `/sync` and have the proper decrementing `stream_ordering` as we import
+        backfilled = False
+        if event.content.get(EventContentFields.MSC2716_HISTORICAL, None) is not None:
```

Comment:
I wonder if we need to be concerned about allowing clients to set this? We should probably only allow this for events that came via bulk send and are authorised to do this?

Reply:
I've updated to use `event.internal_metadata.is_historical()` :+1: